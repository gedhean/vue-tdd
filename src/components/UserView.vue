<template>
  <div>
    User View
    <user-search-form />
    <user-profile :user="user" />
  </div>
</template>

<script>
import UserProfile from "@/components/UserProfile.vue";
import UserSearchForm from "@/components/UserSearchForm.vue";

const components = {
  UserProfile,
  UserSearchForm
};

export default {
  name: "UserView",
  components,
  data() {
    return {
      user: {}
    };
  }
};
</script>

<style></style>
