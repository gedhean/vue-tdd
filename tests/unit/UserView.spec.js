import { shallowMount } from "@vue/test-utils";
import UserView from "@/components/UserView.vue";

let wrapper;
const mountComponent = options => shallowMount(UserView, options);

describe("UserView", () => {
  beforeEach(() => {
    wrapper = mountComponent();
  });

  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("name should be UserView", () => {
    expect(wrapper.name()).toBe("UserView");
  });
});
