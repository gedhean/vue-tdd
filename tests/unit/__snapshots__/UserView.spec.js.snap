// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView should render 1`] = `
<div>
  User View
</div>
`;
